frontend/src/components/RecordsList.tsx

```diff
@@ -175,25 +191,28 @@ export default function RecordsList({
     <Container fluid>
       {showDownloadModal && (
         <DownloadModal
-          loader={() => {
-            return fetchMore({
+          loader={async () => {
+            const sortModel = getSortModel();
+
+            const { data } = await fetchMore({
               variables: {
                 where: {
                   OR: queryFilterWhereVariables(parsedSearchVals),
                 },
                 options: {
                   offset: 0,
                   limit: undefined,
-                  sort: defaultSort,
+                  sort: sortModel,
                 },
               },
-            }).then(({ data }) => {
-              let agGridData = data;
-              if (prepareDataForAgGrid) {
-                agGridData = prepareDataForAgGrid(data, filterModel);
-              }
-              return buildTsvString(agGridData[dataName], colDefs);
             });
+
+            let agGridData = data;
+            if (prepareDataForAgGrid) {
+              agGridData = prepareDataForAgGrid(data, filterModel, sortModel);
+            }
+
+            return buildTsvString(agGridData[dataName], colDefs);
```

Comment:
tested this myself and confirmed that the records are saved to the file in the order in which they're displayed. thanks!